packages/react-ui/components/CurrencyInput/CurrencyInput.tsx

```diff
@@ -114,7 +114,10 @@ export class CurrencyInput extends React.PureComponent<CurrencyInputProps, Curre
 
   public componentDidUpdate(prevProps: CurrencyInputProps, prevState: CurrencyInputState) {
     const { value, fractionDigits, hideTrailingZeros } = this.props;
-    if (value !== CurrencyHelper.parse(prevState.formatted) || prevProps.fractionDigits !== fractionDigits) {
+    if (
+      CurrencyHelper.parse(`${value}`) !== CurrencyHelper.parse(prevState.formatted) ||
```

Comment:
```suggestion
      CurrencyHelper.parse(value?.toString() ?? '') !== CurrencyHelper.parse(prevState.formatted) ||
```

Кажется, что с такой реализацией могут возникнуть проблемы, так как она преобразует `null` и `undefined` в строки:
<img width="152" alt="Снимок экрана 2022-05-08 в 15 01 33" src="https://user-images.githubusercontent.com/48599460/167293164-945970b0-1acc-4947-b097-12c61f3d0adb.png">

Думаю, что тут лучше использовать `value?.toString() ?? ''`, так как `toString()` будет преобразовывать `null` и `undefined` в `null`:
<img width="183" alt="Снимок экрана 2022-05-08 в 15 02 39" src="https://user-images.githubusercontent.com/48599460/167293215-9896132e-bb47-4a80-8439-058c10bfb43b.png">